rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -542,7 +574,37 @@ impl Handler<TaskResult<PartitionOutput, PartitionError>> for CompactOrchestrato
             Some(recs) => recs.records,
             None => todo!(),
         };
-        self.write(records, ctx).await;
+        self.materialize_log(records, ctx.receiver(), ctx).await;
+    }
+}
+
+#[async_trait]
+impl Handler<TaskResult<MaterializeLogsResult, MaterializeLogOperatorError>>
+    for CompactOrchestrator
+{
+    type Result = ();
+
+    async fn handle(
+        &mut self,
+        message: TaskResult<MaterializeLogsResult, MaterializeLogOperatorError>,
+        ctx: &crate::system::ComponentContext<CompactOrchestrator>,
+    ) {
+        let materialized_result = match self.ok_or_terminate(message.into_inner(), ctx) {
+            Some(result) => result,
+            None => return,
+        };
+
+        if materialized_result.is_empty() {
```

Comment:
I am a little confused by this branch, why do we remove a write task if the result is empty? Isn't these one materialization result, and then the write tasks would have to all be removed if its empty?

Reply:
There are N partitions, so there are N write tasks (a "write task" being a `WriteSegments` task). If the materialization result is empty for a given partition, we skip the write task for that partition.

Reply:
ah ok yep got it, the write tasks are set to n before we actually finish materialization